fedimint-server/src/config.rs

```diff
@@ -83,7 +89,13 @@ pub struct ServerConfigConsensus {
     /// Network addresses and names for all peer APIs
     pub api: BTreeMap<PeerId, ApiEndpoint>,
     /// All configuration that needs to be the same for modules
-    pub modules: BTreeMap<String, serde_json::Value>,
+    pub modules: BTreeMap<ModuleInstanceId, JsonWithKind>,
+}
+
+impl ServerConfigConsensus {
+    pub fn module_kinds_iter(&self) -> impl Iterator<Item = (ModuleInstanceId, &ModuleKind)> + '_ {
```

Comment:
Naming again, while it includes the kind, the thing it's iterating over is all module instances.